### ISSUE TITLE
New products do not show anymore on the main page

### ISSUE THREAD

**Issue by jwlalos** [woocommerce/woocommerce#15638]
### EXPLANATION OF THE ISSUE

Woocommerce version 3.0.8 

### STEPS TO REPRODUCE THE ISSUE

1. I take the product which I put online before updating to 3.0.8
2. I duplicate it
3. I change photos and quantities, and description
4. I publish it 

### SYSTEM STATUS

<details>
```
`
### WordPress Environment ###

Home URL: http://www.altaia.eu
Site URL: http://www.altaia.eu
WC Version: 3.0.8
Log Directory Writable: ✔
WP Version: 4.8
WP Multisite: –
WP Memory Limit: 640 MB
WP Debug Mode: –
WP Cron: ✔
Language: en_US

### Server Environment ###

Server Info: Apache
PHP Version: 7.0.7
PHP Post Max Size: 300 MB
PHP Time Limit: 60
PHP Max Input Vars: 1000
cURL Version: 7.26.0
OpenSSL/1.0.1t

SUHOSIN Installed: –
MySQL Version: 5.6.33
Max Upload Size: 300 MB
Default Timezone is UTC: ✔
fsockopen/cURL: ✔
SoapClient: ✔
DOMDocument: ✔
GZip: ✔
Multibyte String: ✔
Remote Post: ✔
Remote Get: ✔

### Database ###

WC Database Version: 3.0.8
WC Database Prefix: wp_103302_
woocommerce_sessions: ✔
woocommerce_api_keys: ✔
woocommerce_attribute_taxonomies: ✔
woocommerce_downloadable_product_permissions: ✔
woocommerce_order_items: ✔
woocommerce_order_itemmeta: ✔
woocommerce_tax_rates: ✔
woocommerce_tax_rate_locations: ✔
woocommerce_shipping_zones: ✔
woocommerce_shipping_zone_locations: ✔
woocommerce_shipping_zone_methods: ✔
woocommerce_payment_tokens: ✔
woocommerce_payment_tokenmeta: ✔
MaxMind GeoIP Database: ✔

### Security ###

Secure connection (HTTPS): ❌Your store is not using HTTPS. Learn more about HTTPS and SSL Certificates.
Hide errors from visitors: ✔

### Active Plugins (23) ###

Polylang: by Frédéric Demarle – 2.1.5
Color Filters for WooCommerce: by Elementous – 1.2.2
Count Per Day: by Tom Braider – 3.5.7
Currency Switcher for WooCommerce: by Algoritmika Ltd – 2.3.1
Elegant Themes Support: by Elegant Themes – 1.3
Force Regenerate Thumbnails: by Pedro Elsner – 2.0.6
Google Analytics Dashboard for WP (GADWP): by Alin Marcu – 5.0.1
oik Weight/Country Shipping: by bobbingwide – 1.3.4
Perfect WooCommerce Brands: by Alberto de Vera Sevilla – 1.4.5
Regenerate Thumbnails: by Alex Mills (Viper007Bond) – 2.2.6
SI Captcha Anti-Spam: by Mike Challis – 3.0.0.19
VOD Infomaniak: by Infomaniak Staff – 1.3.11
WooCommerce Product Feed: by WebAppick – 2.1.6
PayU Payment Gateway: by PayU SA – 1.2.1
Hyyan WooCommerce Polylang Integration: by Hyyan Abo Fakher – 1.0.1
WooCommerce Cart Tab: by jameskoster – 1.1.1
WooCommerce Stripe Gateway: by WooCommerce – 3.1.9
WooCommerce: by Automattic – 3.0.8
WordPress HTTPS: by Mike Ems – 3.3.6
WordPress Importer: by wordpressdotorg – 0.6.3
Yoast SEO: by Team Yoast – 4.9
WP Rollback: by WordImpress – 1.4
WP Super Cache: by Automattic – 1.4.9

### Settings ###

API Enabled: ✔
Force SSL: –
Currency: PLN (zł)
Currency Position: right_space
Thousand Separator: ,
Decimal Separator: .
Number of Decimals: 2
Taxonomies: Product Types: external (external)
grouped (grouped)
simple (simple)
variable (variable)

Taxonomies: Product Visibility: exclude-from-catalog (exclude-from-catalog)
exclude-from-search (exclude-from-search)
featured (featured)
outofstock (outofstock)
rated-1 (rated-1)
rated-2 (rated-2)
rated-3 (rated-3)
rated-4 (rated-4)
rated-4 (rated-4-2)
rated-5 (rated-5)


### WC Pages ###

Shop base: ❌ Page not set
Cart: #1148 - /pl/koszyk/
Checkout: #1163 - /pl/kontrola-zamowienia/
My account: #1194 - /pl/logowanie-rejestracja/

### Theme ###

Name: eStore
Version: 5.1.9
Author URL: http://www.elegantthemes.com
Child Theme: ❌ – If you're modifying WooCommerce on a parent theme you didn't build personally
then we recommend using a child theme. See: How to create a child theme

WooCommerce Support: ✔

### Templates ###

Overrides: –
`
```
</details>

**Comment by mikejolley** [comment]
You're using caching plugins - you'll want to flush the caches if this is being done aggressively. 

If only duplicated products are affected, it may be stock status. In which case, wait for 3.0.9 where there is a fix for these edge cases. https://github.com/woocommerce/woocommerce/pull/15569

**Comment by jwlalos** [comment]
Hi, Mike. Thank you for your prompt answer. Not only duplicated products are affected. All products. I added a product from scratch; the same: shows in the product category (Mini), but not on the main page. Looks like there shuld be a function "show the product on the main page" +"show the product in the assigned category". I cleared the cache, I switched off the WP cache plugin, no difference. 
Please, don't let me hang dry here, I try to find a solution since 2 weeks.

**Comment by mikejolley** [comment]
Try a different theme.

**Comment by jwlalos** [comment]
Tried twenty storefrront and other themes from elegant themes. Nothing works. 
The only way I found out to show the added products finally is to roll backk the woocommerce to 2.6, duplicate the products which didn't whow up i 3.0.8, update to 3.0.8 again and publish them. 
But this is a quite drastic and cumbersome measure.

**Comment by mikejolley** [comment]
There is also a setting in Settings > Products > Inv. Stop hiding out of stock products - does that help?

**Comment by jwlalos** [comment]
Done right now. No influence on the visibility of the products on the first page.

**Comment by mikejolley** [comment]
> Hyyan WooCommerce Polylang Integration

Disabled that?

You're supposed to do these types of checks before logging issues FYI. Conflict identification is step 1 :)

**Comment by jwlalos** [comment]
I just switched off all plugins except woocommerce. I start over again. No products on the first page 404 error. I will be switching one by one. I will back to you after my "big test of everything".

**Comment by mikejolley** [comment]
Right, test with a default theme, WC only, and re-save permalinks, flush cache, before continuing.

Enable the caching plugin, flush cache, disable caching plugin to be safe.

**Comment by mikejolley** [comment]
O important

> Shop base: ❌ Page not set

If you're not viewing the shop page, how can you be sure you're seeing output from WC?

**Comment by jwlalos** [comment]
I asked myself the same question, as it seemed strange. Now I am at the point that only Woocommerce plugin is active and Polylang. Looks like Polylang governs the whole visibility :-O. I switch off the Polylang plugin - error 404. I switch on - products are visible. No logic in that to me, but I have to agree on that solution, if I want to have shop working.

**Comment by jwlalos** [comment]
I will share what I found switching between WC 3.0.8 and WC 2.6.14: once I duplicate the product, even if I publish it, it doesn't show up on the main page.
BUT.
If I go to the Quick Edit (in the 2.6.14 version, I see that the visibility is set up by default to Hidden.
I changed it to "Visible", product appeared on the main page. 
I tried to do the same operation with the WC 3.0.8, but unfortunately this function doesn't appear in te "Quick edit".
My wild guess is that in WC 3.0.8 all products added get automatically status "Hidden". which is not accessible to the shop manager to change. 
Well, looks like I will be rolling the versions back and forth till the next release with (HOPEFULLY) fixed problem of products' apperance. 
In any case - thank you for spending some time with me and your suggestions. 
Justyna
